sysdump/sysdump.go

```diff
@@ -0,0 +1,815 @@
+				defer s.tasksWg.Done()
+				p, err := s.client.ListPods(ctx, s.options.HubbleRelayNamespace, metav1.ListOptions{
+					LabelSelector: s.options.HubbleRelayLabelSelector,
+				})
+				if err != nil {
+					return fmt.Errorf("failed to get logs from Hubble Relay pods")
+				}
+				if err := s.submitLogsTasks(ctx, filterPods(p, nodeList), sinceTime, s.options.LogsLimitBytes, absoluteTempPath); err != nil {
+					return fmt.Errorf("failed to collect logs from Hubble Relay pods")
+				}
+				return nil
+			},
+			// Get logs from Hubble UI.
+			func(ctx context.Context) error {
+				s.logTask("Collecting logs from Hubble UI pods")
+				defer s.logDebug("Finished collecting logs from Hubble UI pods")
+				defer s.tasksWg.Done()
+				p, err := s.client.ListPods(ctx, s.options.HubbleNamespace, metav1.ListOptions{
+					LabelSelector: s.options.HubbleLabelSelector,
+				})
+				if err != nil {
+					return fmt.Errorf("failed to get logs from Hubble UI pods")
+				}
+				if err := s.submitLogsTasks(ctx, filterPods(p, nodeList), sinceTime, s.options.LogsLimitBytes, absoluteTempPath); err != nil {
+					return fmt.Errorf("failed to collect logs from Hubble UI pods")
+				}
+				return nil
+			},
+		}...)
+	}
+
+	// Add the tasks to the worker pool.
+	for i, t := range tasks {
+		t := t
+		s.tasksWg.Add(1)
+		if err := s.pool.Submit(fmt.Sprintf("%d", i), t); err != nil {
+			return fmt.Errorf("failed to submit task to the worker pool: %w", err)
+		}
+	}
+
+	// Wait for the tasks to finish.
+	s.tasksWg.Wait()
+	var merr error
+	r, err := s.pool.Drain()
+	if err != nil {
+		return fmt.Errorf("failed to drain the worker pool :%w", err)
+	}
+	for _, res := range r {
+		if err := res.Err(); err != nil {
+			merr = multierror.Append(merr, err)
+		}
+	}
+
+	// Close the worker pool.
+	if err := s.pool.Close(); err != nil {
+		return fmt.Errorf("failed to close the worker pool: %w", err)
+	}
+
+	// Check if any errors occurred and warn the user.
+	if merr != nil {
+		s.logWarn("The sysdump may be incomplete — %s", merr.Error())
+		s.logWarn("Please note that depending on your Cilium version and installation options, this may be expected")
+	}
+
+	// Create the zip file in the current directory.
+	s.log("🗳 Compiling sysdump")
+	p, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("failed to get current directory: %w", err)
+	}
+	f := filepath.Join(p, replaceTimestamp(s.options.OutputFileName)+".zip")
+	if err := archiver.Archive([]string{d}, f); err != nil {
+		return fmt.Errorf("failed to create zip file: %w", err)
+	}
+	s.log("✅ The sysdump has been saved to %s", f)
+
+	// Try to remove the temporary directory.
+	s.logDebug("Removing the temporary directory %s", d)
+	if err := os.RemoveAll(d); err != nil {
+		s.logWarn("failed to remove temporary directory %s: %v", d, err)
+	}
+	return nil
+}
+
+func (s *sysdump) log(msg string, args ...interface{}) {
+	fmt.Fprintf(s.options.Writer, msg+"\n", args...)
```

Comment:
Timestamps may be nice here, but I'm not sure if we're intentionally keeping parity with previous output

Reply:
It's more of keeping parity with current output of the CLI — I stole this from somewhere else. However, I agree that logging should be made uniform throughout (of course, as part of a separate PR 😛).